src/collections.h

```diff
@@ -238,6 +249,62 @@ Collection<T>::map(Function func) {
 };
 
 
+template<typename T, typename Function>
+void *
+conmap_thread(void *arg) {
+    struct thread_data {
+        std::vector<T> *list;
+        Function func;
+        int begin;
+        int end;
+    };
+    thread_data *data = ((struct thread_data *)arg);
+    for (int i = data->begin; i < data->end && i < (*(data->list)).size(); i++)
+        (*(data->list))[i] = (data->func)((*(data->list))[i]);
+    pthread_exit(NULL);
+};
+
+
+template<typename T>
+template<typename Function>
+Collection<typename std::result_of<Function(T)>::type>
+Collection<T>::conmap(Function func, int threads) {
```

Comment:
I think this should be named `pmap` since it's in parallel, not concurrent.
When we rewrite it to use C++'s async functions it can be `conmap`.